### ISSUE TITLE
[Old Firefox - version 45] Syntax error in strict mode

### ISSUE THREAD

**Issue by Aswathprabhu** [ionic-team/stencil#2839]
**Stencil version:**
```
@stencil/core@1.17.3
```

**I'm submitting a:**

- [x] bug report

**Current behavior:**
Trying to load the nomodule file on an old browser will throw a warning in the console that prevents the entire file from being executed:

> SyntaxError: in strict mode code, functions may be declared only at top level or immediately within another function

**Expected behavior:**
Should be working normally

**User Agent**

> Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:45.0) Gecko/20100101 Firefox/45.0

**Comment by chenyulun** [comment]
#2755

**Comment by chenyulun** [comment]
@stencil/core 2.5.0
```js
extras: {
    appendChildSlotFix: true,
    cloneNodeFix: true,
    slotChildNodesFix: false,
    cssVarsShim: true,
    dynamicImportShim: true,
    initializeNextTick: true,
    safari10: true,
    scriptDataOpts: true,
    shadowDomShim: true,
  },
```
This version still has this problem

**Comment by rwaskiewicz** [comment]
I apologize that this is the first you're seeing any official response from the team. As Firefox was released over 6 years ago (2016.03.08), this falls outside of the scope of the browsers that we're able to support. As a result, I'm going to close this issue. Thanks again!
